prometheus/base/alerting/prometheus.argocd_alerts.yaml

```diff
@@ -17,3 +17,14 @@ spec:
       annotations:
         summary: "Application {{ $labels.name }} in namespace {{ $labels.dest_namespace }} was 'Degraded'"
         description: "Application {{ $labels.name }} in namespace {{ $labels.dest_namespace }} was 'Degraded' in the last 5 minutes"
+    - alert: ProgressingArgocdApp
+      expr: |
+        max_over_time(argocd_app_info{health_status="Progressing", dest_namespace!~".+-tenant"}[20m]) == 1
+        and ignoring (health_status)
+        max_over_time(argocd_app_info{health_status!="Healthy", dest_namespace!~".+-tenant"}[4m]) == 1
```

Comment:
We have status healthy mentioned on the tests quite a lot, but the expression is no longer considering it. We should have the tests series mention other statuses which are not progressing or healthy.

Reply:
It is now changed, now "healthy" health status is mentioned only where it's still relevant

Reply:
In which cases is it relevant? The alert expression is not considering metrics with status=healthy.

Reply:
We should instead have test cases that has metrics with labels which are not healthy or progressing in the last 4 minutes so we test the part of the expression that matches !=healthy.

Reply:
Now the only comments with the word "Healthy" or it's variations are:
1.       # app was progessing and then Healthy for the last 5 min, should not alert
2.       # app was progessing and then Healthy for the last 3 min, should alert
3.       # the app is switching health states, alert should be triggered

which of them does not comply with these rules?

Reply:
There's this line:
``` max_over_time(argocd_app_info{health_status!="Healthy", dest_namespace!~".+-tenant"}[4m]) == 1``` 
so in order for the tests to be clear and understandable there's a need to state whether the app is healthy and if so for how long, this is just in my opinion .

Reply:
The thing is, the tests will pass just the same if you delete all series with status=healthy. I think that adding the healthy series in comment or to the first use case just to convey the idea might be helpful. I don't think we should have it on all test cases while the rule is ignoring it.

Reply:
ok, I removed it